### ISSUE TITLE
Cilium agents lost endpoint connectivity after rolling restart

### ISSUE THREAD

**Issue by wimnat** [cilium/cilium#27922]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### What happened?

**What we were doing**

A Cilium config parameter was changed (setting `idleConnectionGracePeriod` to 60s) and this triggered a rolling restart of Cilium agents.

This change was performed across around 30 clusters all running the same k8s version and Cilium version.

**What happened**

On our largest cluster (50+ nodes), immediately after the agent restart, we starting to see an uptick in `cilium_unreachable_health_endpoints` count metric across multiple nodes in the cluster.  Pods running on those nodes lost network connectivity.

It was resolved by restarting the Cilium agent.  In some cases the agent had to be restarted multiple times.

All the other clusters we deployed to did not exhibit this problem.

A sysdump from two bad nodes was retrieved.

`cilium status` output from one particular node:

```
==== detail from pod cilium-q8scs , on node ip-10-1-229-163.ec2.internal
KVStore:                Ok   Disabled
Kubernetes:             Ok   1.25+ (v1.25.12-eks-2d98532) [linux/amd64]
Kubernetes APIs:        ["cilium/v2::CiliumClusterwideNetworkPolicy", "cilium/v2::CiliumEndpoint", "cilium/v2::CiliumNetworkPolicy", "cilium/v2::CiliumNode", "core/v1::Namespace", "core/v1::Node", "core/v1::Pods", "core/v1::Service", "discovery/v1::EndpointSlice", "networking.k8s.io/v1::NetworkPolicy"]
KubeProxyReplacement:   Disabled
Host firewall:          Disabled
CNI Chaining:           none
CNI Config file:        CNI configuration file management disabled
Cilium:                 Ok   1.13.5 (v1.13.5-9b8fa411)
NodeMonitor:            Listening for events on 32 CPUs with 64x4096 of shared memory
Cilium health daemon:   Ok
IPAM:                   IPv4: 13/32 allocated,
IPv6 BIG TCP:           Disabled
BandwidthManager:       Disabled
Host Routing:           Legacy
Masquerading:           IPTables [IPv4: Enabled, IPv6: Disabled]
Controller Status:      70/71 healthy
  Name                    Last success   Last error    Count   Message
  ipcache-inject-labels   never          2h6m12s ago   112     k8s cache not fully synced
Proxy Status:                      OK, ip 10.145.231.136, 29 redirects active on ports 10000-20000
Global Identity Range:             min 256, max 65535
Hubble:                            Ok   Current/Max Flows: 102/4095 (2.49%), Flows/s: 0.01   Metrics: Ok
Encryption:                        Disabled
Cluster health:                    28/54 reachable   (2023-09-04T08:19:25Z)
  Name                             IP                Node          Endpoints
  ip-10-1-224-49.ec2.internal    10.1.224.49     reachable     unreachable
  ip-10-1-224-82.ec2.internal    10.1.224.82     unreachable   unreachable
  ip-10-1-225-117.ec2.internal   10.1.225.117    unreachable   unreachable
  ip-10-1-225-192.ec2.internal   10.1.225.192    unreachable   unreachable
  ip-10-1-225-242.ec2.internal   10.1.225.242    reachable     unreachable
  ip-10-1-226-192.ec2.internal   10.1.226.192    unreachable   unreachable
  ip-10-1-226-193.ec2.internal   10.1.226.193    unreachable   unreachable
  ip-10-1-226-37.ec2.internal    10.1.226.37     reachable     unreachable
  ...
```

**What I expected to happen**

Cilium agent rolling restart completed without impacting any existing traffic flows and no increase in unhealthy endpoints metric

### Cilium Version

Cilium:                 Ok   1.13.5 (v1.13.5-9b8fa411)

### Kernel Version

Linux ip-10-1-228-145 5.15.0-1040-aws #45~20.04.1-Ubuntu SMP Tue Jul 11 19:08:13 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux

### Kubernetes Version

v1.25.12-eks-2d98532

### Sysdump

Can be shared privately

### Relevant log output

```shell
Agent logs were unremarkable
```

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by youngnick** [comment]
Just to be clear, the rollout worked across 29 of the 30 clusters, but required additional Cilium agent restarts on the largest cluster?

Is there anything else different about that cluster aside from the size?

**Comment by wimnat** [comment]
Yep that's right @youngnick.  The rollout was successful across all other clusters.  No, there is no difference between them other than size - same k8s control plane version, same Cilium version, same node images so same kernel.

**Comment by wimnat** [comment]
@youngnick would it be useful if i sent you the sysdumps?

**Comment by christarazi** [comment]
Given the version is v1.13.5, I think you're hitting the same as https://github.com/cilium/cilium/issues/26295, which is fixed in v1.13.6.

**Comment by wimnat** [comment]
We rolled out `1.13.7` yesterday and we did not experience the same issue when deploying to the large cluster.  I agree that I think this issue is a duplicate of https://github.com/cilium/cilium/issues/26295.  Closing this - thanks for the help! 🙇
